src/main/com/cw/converter/GalacticCodeToDecimalConverter.java

```diff
@@ -0,0 +1,61 @@
+package main.com.cw.converter;
+
+import main.com.cw.util.RomanCharToDecimalConverter;
+import main.com.cw.validator.GalacticCodeFormatValidator;
+
+/**
+ * Created by cwijayasundara on 23/01/2017.
+ * This class implement the generic NumberConverter interface and override the convert() to
+ * convert a galactic code to the curresponding decimal / arabic number
+ */
+public class GalacticCodeToDecimalConverter implements NumberConverter {
+
+    private GalacticCodeFormatValidator galacticCodeFormatValidator;
+
+    private RomanCharToDecimalConverter romanCharToDecimalConverter;
+
+    public GalacticCodeToDecimalConverter(){
+        romanCharToDecimalConverter = new RomanCharToDecimalConverter();
+        galacticCodeFormatValidator = new GalacticCodeFormatValidator();
+    }
+
+    /* convert the given galactic code to curresponding decimal value
+     *
+     */
+    public int convert(final String numberToConvert){
+
+        String galacticString = numberToConvert.toUpperCase();
+        int galacticSum=0;
+        int counter=0;
+        int galacticeValueLength = galacticString.length();
+
+        if (galacticCodeFormatValidator.isValidFormat(galacticString)){
+            // loop if value(n+1) > value(n) then value(n+1) - value(n)
+            while(counter< galacticeValueLength){
```

Comment:
Minor: now a days we rarely use while loops and I would have opted for a Char stream and map here, perhaps. Its only a syntactic suger suggestion than any issue with algorithmic change.